Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -3627,6 +3669,295 @@ public bool RemoveTransferredSubscription(Subscription subscription)
         }
         #endregion
 
+        #region Combined Browse/BrowseNext
+
+        /// <summary>
+        /// Execute browse and, if necessary, browse next in one
+        /// service call. Take care of BadNoContinuationPoint and BadInvalidContnuationPoint errors
+        /// </summary>
+        /// <param name="requestHeader"></param>
+        /// <param name="view"></param>
+        /// <param name="nodesToBrowse"></param>
+        /// <param name="maxResultsToReturn"></param>
+        /// <param name="browseDirection"></param>
+        /// <param name="referenceTypeId"></param>
+        /// <param name="includeSubtypes"></param>
+        /// <param name="nodeClassMask"></param>
+        /// <param name="excecuteDefensively"></param>
+        /// <param name="result"></param>
+        /// <param name="errors"></param>
+        /// <returns></returns>
+        public void ManagedBrowse(
```

Comment:
I would like to see the following additions: 
- overload in Traceable Session
- use in FetchReferences.
- async variant of the method.
- use in referenceClient (browsefulladressspace)

Reply:
That's work in progress. Do you think I should add it to ISession, as well?

Reply:
- added the method to ISession
- added Async version (to ISession and SessionAsync)
- added support in Traceable session
- use the method (as of now controlled by a parameter optionally) in FetchReferences (both versions) and FetchReferencesAsynd
- created a sample method call in the console reference client ClientSamples
- enhanced the unit tests to test the FetchReferences methods (indirectly through the node cache)
- added unit tests to test the ManagedBrowseAsync Method.

Reply:
coming together nicely, great work!